opm/autodiff/AquiferCarterTracy.hpp

```diff
@@ -0,0 +1,443 @@
+/*
+  Copyright 2017 TNO - Heat Transfer & Fluid Dynamics, Modelling & Optimization of the Subsurface
+  Copyright 2017 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_AQUIFERCT_HEADER_INCLUDED
+#define OPM_AQUIFERCT_HEADER_INCLUDED
+
+#include <opm/parser/eclipse/EclipseState/AquiferCT.hpp>
+#include <opm/parser/eclipse/EclipseState/Aquancon.hpp>
+#include <opm/autodiff/BlackoilAquiferModel.hpp>
+#include <opm/common/OpmLog/OpmLog.hpp>
+#include <opm/common/utility/numeric/linearInterpolation.hpp>
+#include <opm/core/props/BlackoilPhases.hpp>
+
+
+#include <opm/material/densead/Math.hpp>
+#include <opm/material/densead/Evaluation.hpp>
+#include <opm/material/fluidstates/BlackOilFluidState.hpp>
+
+#include <string>
+#include <memory>
+#include <vector>
+#include <algorithm>
+#include <map>
+#include <cassert>
+
+namespace Opm
+{
+
+    template<typename TypeTag>
+    class AquiferCarterTracy
+    {
+
+        public:
+            typedef BlackoilModelParameters ModelParameters;
+
+            static const int Water = BlackoilPhases::Aqua;
+            static const int Oil = BlackoilPhases::Liquid;
+            static const int Gas = BlackoilPhases::Vapour;
+
+            typedef typename GET_PROP_TYPE(TypeTag, Grid) Grid;
```

Comment:
This class requires some substantial cleaning up. Basically, I guess the Oil, Gas, and polymer and solvent related are not needed in the current pull request, please remove them.